src/compiler/types.ts

```diff
@@ -5099,17 +5099,19 @@ export interface TypeChecker {
     getBaseConstraintOfType(type: Type): Type | undefined;
     getDefaultFromTypeParameter(type: Type): Type | undefined;
 
-    /** @internal */ getAnyType(): Type;
-    /** @internal */ getStringType(): Type;
-    /** @internal */ getNumberType(): Type;
-    /** @internal */ getBooleanType(): Type;
-    /** @internal */ getFalseType(fresh?: boolean): Type;
-    /** @internal */ getTrueType(fresh?: boolean): Type;
-    /** @internal */ getVoidType(): Type;
-    /** @internal */ getUndefinedType(): Type;
-    /** @internal */ getNullType(): Type;
-    /** @internal */ getESSymbolType(): Type;
-    /** @internal */ getNeverType(): Type;
+    getAnyType(): Type;
+    getStringType(): Type;
+    getStringLiteralType(value: string): StringLiteralType;
+    getNumberType(): Type;
+    getNumberLiteralType(value: number): NumberLiteralType;
+    getBooleanType(): Type;
+    getFalseType(fresh?: boolean): Type;
+    getTrueType(fresh?: boolean): Type;

```

Comment:
Rereading this again, do we want to expose the fresh parameter? Or should we maybe overload it and only expose one without parameters publicly?

Reply:
What does the fresh param do?  In my personal needs using this, I'm just invoking `getFalseType()`.  How likely is it that people using this would need/want to pass in the fresh arg?

Reply:
Literal types have fresh and regular variants. When assigning from a fresh literal to a mutable location, the resulting type is widened to the base type:

```ts
const freshFalse = false;
const regularFalse: false = false;

let widenedToBoolean = freshFalse;
let stillFalse = regularFalse;
```

from an API perspective, this means it’s important to note that

```ts
checker.getFalseType(true) !== checker.getFalseType(false)
```

but that a `false` type will always be _one_ of these. Within the checker, we would probably never do an equality comparison to determine if a type is `false`; we’d call `getTypeFacts` and look at those, but that’s not exported. Soooo, whether to expose it, 🤷‍♂️?

Reply:
Lots of places appear to do things like `foo === falseType || foo === freshFalseType` too. I guess it depends, but it might be important if you are trying to do something with these functions...

Alternatively what people want externally is "is this assignable to false" but obviously that API is a different issue so...

Reply:
Interesting.  So in my case, I have `checker.getFalseType(false)` (since I don't pass anything in).  And I'm later using that, but not with `===`.  Rather, I'm using it with `isTypeAssignableTo`, because what I desire to know is if some type I have could accept the value `false`.

For example:

```typescript
const foo: false | number = getRandomNumberOrMaybeFalse();
```
Later, I want to know if `foo` would accept `false`. I don't want or need to know if `foo` would accept `boolean`, only if it will accept `false`.  

But I have to admit that I don't know which of the two: `checker.getFalseType(true)` or `checker.getFalseType(false)` I want in this case, or if it even matters?

And the whole reason I mention my use-case at all is to try to add insight that might help decide if we need to expose the `freshType` arg or not?  I can't purport to know what others might be doing with `getFalseType` or if their use-cases are at all like mine.

Reply:
Fresh/regular shouldn’t matter for assignability, so it sounds like in your case it doesn’t matter.

Noticing that a couple of our codefixes grab a type, check its flags for `BooleanLiteral`, and then compare to `checker.getFalseType(true)` and `checker.getFalseType(false)`, I’m inclined to say it’s fine to leave it exposed. Codemods are a good example of a reasonable usage of our API.

Reply:
Obviously the value of my input is limited, since I don't have the same depth on knowledge about TS that you guys have, but my intuition would be to leave it exposed.  As a developer, I prefer having flexibility available to me, even if I don't use it, so long as that flexibility doesn't make my life difficult.  And in this case, it doesn't seem to, so I say leave it.